### DISCUSSION TITLE
Using Diamond for looking up very short DNA sequences

### DISCUSSION THREAD

**Discussion by fluhus** [bbuchfink/diamond#427] [Q&A] [answered]
Hi,

I hope this is an appropriate place for my question :)

I have been using Diamond for a while in my research I find it useful and friendly.
Now I am facing a new challenge, to look up 32-mers of DNA in a protein database. **However, just running with default parameters returns 0 alignments.** I assume the mappings get thrown away for low confidence or something like that(?).

**How can I make Diamond still print whatever alignments it finds?**

My command so far (not sure what I am doing, I tried to reduce every threshold to 0 to make it print out _anything_):

```
./diamond blastx -d uniref50 -q kmers.fa.gz -f 101 -o kmers.diamond.sam -p 10 -e 1 -k 2 --subject-cover 0 -l 0 --query-cover 0 --id 0 --id2 0 --max-hsps 0
```

I am still getting: `0 queries aligned.`

Thanks,
Amit

**Comment by bbuchfink** [comment, marked as answer]
Try using `--very-sensitive -s1 -l1 -e 10000`.

`--short-query-ungapped-bitscore 15 --gapped-filter-evalue 0` might also help, but you need to compile with `cmake -DEXTRA=ON` to activate these options.

**Reply by fluhus** [reply]
Thank you!

I tried the first option and it did manage to map about 6% of the kmers.

I am now testing the second option. Got compilation errors, see below (GCC 4.8.5). I managed to fix them by calling the explicit constructor: `return std::tuple<char*, size_t, int>( ... );`.

```
src/util/system/system.cpp:174:27: error: converting to \u2018std::tuple<char*, long unsigned int, int>\u2019 from initializer list would use explicit constructor \u2018constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {std::nullptr_t, int, int}; <template-parameter-2-2> = void; _Elements = {char*, long unsigned int, int}]\u2019
   return { nullptr, 0, -1 };
                           ^
src/util/system/system.cpp:185:35: error: converting to \u2018std::tuple<char*, long unsigned int, int>\u2019 from initializer list would use explicit constructor \u2018constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {char*, long unsigned int&, int&}; <template-parameter-2-2> = void; _Elements = {char*, long unsigned int, int}]\u2019
  return { (char*)addr, length, fd };
                                   ^
```

**Reply by fluhus** [reply]
To follow up, I used the second option and it gave me what I needed.

Thank you!
